src/hyperv/hyperv_network_api_v2.c

```diff
@@ -71,103 +71,141 @@ hyperv2MsvmVirtualSwitchToNetwork(virConnectPtr conn,
+    Msvm_VirtualEthernetSwitch_V2 *vSwitch = NULL;
+    virNetworkPtr network = NULL;
+    virNetworkPtr *nets = NULL;
+    int count = 0;
+    int ret = -1;
+    size_t nnets = 0;
+    size_t i = 0;
+
+    virCheckFlags(VIR_CONNECT_LIST_NETWORKS_FILTERS_ALL, -1);
+
+    /* filter out flag options that will produce 0 results:
+     * - inactive: all networks are active
+     * - transient: all networks are persistend
+     * - no autostart: all networks are auto start
+     */
+    if ((MATCH(VIR_CONNECT_LIST_NETWORKS_INACTIVE) &&
+         !MATCH(VIR_CONNECT_LIST_NETWORKS_ACTIVE)) ||
+        (MATCH(VIR_CONNECT_LIST_NETWORKS_TRANSIENT) &&
+         !MATCH(VIR_CONNECT_LIST_NETWORKS_PERSISTENT)) ||
+        (MATCH(VIR_CONNECT_LIST_DOMAINS_NO_AUTOSTART) &&
+         !MATCH(VIR_CONNECT_LIST_NETWORKS_AUTOSTART))) {
+        if (networks && VIR_ALLOC_N(*networks, 1) < 0)
+            goto cleanup;
 
-    virBufferAddLit(&query, MSVM_VIRTUALETHERNETSWITCH_V2_WQL_SELECT);
-    virBufferAsprintf(&query, " where HealthState = 5");
-    if (hyperv2GetMsvmVirtualEthernetSwitchList(priv, &query, &virtualSwitchList) < 0)
+        ret = 0;
         goto cleanup;
+    }
 
-    for (virtualSwitch = virtualSwitchList; virtualSwitch != NULL;
-         virtualSwitch = virtualSwitch->next) {
-        count++;
+    if (hyperv2GetVirtualSwitchList(priv, NULL, &vSwitchList) < 0)
+        goto cleanup;
+
+    if (networks) {
+        if (VIR_ALLOC_N(nets, 1) < 0)
+            goto cleanup;
+        nnets = 1;
     }
 
-    result = count;
+    for (vSwitch = vSwitchList; vSwitch != NULL; vSwitch = vSwitch->next) {
+        /* just count the networks */
+        if (!networks) {
+            count++;
+            continue;
+        }
+
+        if (VIR_RESIZE_N(nets, nnets, count, 2) < 0)
+            goto cleanup;
+
+        if (hyperv2MsvmVirtualSwitchToNetwork(conn, vSwitch, &network) < 0)
+            goto cleanup;
+
+        nets[count++] = network;
+        network = NULL;
+    }
+
+    if (nets)
+        *networks = nets;
+    nets = NULL;
+    ret = count;
 
  cleanup:
-    hypervFreeObject(priv, (hypervObject *) virtualSwitchList);
-    virBufferFreeAndReset(&query);
-    return result;
+    if (nets) {
+        for (i = 0; i < count; ++i)
+            virObjectUnref(nets[i]);
+
+        count = -1;
+    }
+
+    hypervFreeObject(priv, (hypervObject *) vSwitchList);
+
+    return ret;
 }
+#undef MATCH
 
 int
-hyperv2ConnectListDefinedNetworks(virConnectPtr conn, char **const names, int maxnames)
+hyperv2ConnectListNetworks(virConnectPtr conn, char **const names, int maxnames)
 {
-    int i, count = 0;
-    bool success = false;
+    Msvm_VirtualEthernetSwitch_V2 *list = NULL, *entry = NULL;
     hypervPrivate *priv = conn->privateData;
-    virBuffer query = VIR_BUFFER_INITIALIZER;
-    Msvm_VirtualEthernetSwitch_V2 *virtualSwitchList = NULL;
-    Msvm_VirtualEthernetSwitch_V2 *virtualSwitch = NULL;
+    int count = 0;
```

Comment:
same as above, its used as return value on success so it should match the function's return type which is int